### ISSUE TITLE
Upgrade WindowsAppSDK dependency version

### ISSUE THREAD

**Issue by KieranDevvs** [dotnet/maui#14107]
### Description

In regards to the issue relating to the WebView2 control freezing when typing too fast: https://github.com/dotnet/maui/issues/12361

The issue has been fixed in a newer dependency version of WindowsAppSDK. Can MAUI please upgrade the references as manually specifying the `PackageReference` (`<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />`) results in problems building with the following errors:
```
error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.Mono.win-arm
error NU1102: Unable to find package Microsoft.NETCore.App.Runtime.Mono.win-arm64
```
Along with this error when attempting to publish via the CLI:
```
C:\Users\Kieran Devlin\.nuget\packages\microsoft.windowsappsdk\1.2.230313.1\buildTransitive\Microsoft.UI.Xaml.Markup.Co
mpiler.interop.targets(559,9): error MSB3073: The command ""C:\Users\Kieran Devlin\.nuget\packages\microsoft.windowsapp
sdk\1.2.230313.1\buildTransitive\..\tools\net5.0\..\net472\XamlCompiler.exe" "obj\release\net7.0-android\\input.json" "
obj\release\net7.0-android\\output.json"" exited with code 1. [C:\Users\Kieran Devlin\Documents\git\sources\App\..\App.csproj::TargetFramework=net7.0-android]
```

### Steps to Reproduce

1) Specify the dependency version that fixes the referenced problem (`<PackageReference Include="Microsoft.WindowsAppSDK" Version="1.2.230313.1" />`)
2) Build once (the first build seems to work fine)
3) Build again, this build & all subsequent builds will fail.

### Link to public reproduction project repository

N/a

### Version with bug

7.0 (current)

### Last version that worked well

7.0 (current)

### Affected platforms

Android, Windows, I was *not* able test on other platforms

### Affected platform versions

N/a

**Comment by drasticactions** [comment]
I believe that's already bumped in Main? https://github.com/dotnet/maui/blob/main/eng/Versions.props#L25

**Comment by jfversluis** [comment]
Yeah I would assume people would want this as a service release for .NET 7?

@mattleibow ?

**Comment by KieranDevvs** [comment]
> <img alt="@GabrielChouinardLetourneau" width="40" height="40" src="https://avatars.githubusercontent.com/u/37672082?s=80&amp;u=1178f5a69e74cee939d3f96824e5c9d3b8033359&amp;v=4">

While this may be a temporary work around, Id still like to see the framework take on the upgrade rather than delegating it to users.

**Comment by mattleibow** [comment]
We are keeping pretty much up to date. However, I think this is .net 7.

**Comment by Xyncgas** [comment]
So .NET 7 doesn't get the fix because it's not .NET 6?

**Comment by Xyncgas** [comment]
Would https://github.com/dotnet/maui/pull/14435 makes it possible for this issue to be addressed further, letting people no longer have to paste the WindowsAPpSSDK package to fix the problem causing MAUI.Blazor unable to use mouse

**Comment by creacity** [comment]
This issue is also necessary for the compilation problem mentioned in [7846](https://github.com/microsoft/microsoft-ui-xaml/issues/7846). WindowsPackage compilation fails when I use an external MAUI library. 

https://github.com/microsoft/microsoft-ui-xaml/issues/7846

**Comment by creacity** [comment]
How is it possible to get any feedback on this or a possible solution date?

**Comment by mattleibow** [comment]
> This is because you're referencing it also in the android and ios  app.
> If you reference it like this, it will work (only for Windows!)
> ```
> <ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.19041.0'">
>    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230331000" />
> </ItemGroup>
> ```

I think this appears to be a bug because you have to do this, but this is as designed. When a new "major" version of wasdk comes out we will take some time before updating. 

Versions 1.2 and 1.3 are considered "major"-ish versions by the wasdk team. So we don't want to go updating everyone as soon as there is a new version.

.NET 7 is based on 1.2 and .NET 8 is based on 1.3. Depending on feedback - such as this issue - we will investigate and consider updating the version that comes with the Maui install. 

Regardless of everything and anything, having to specify the updated version of wasdk is a feature. Allowing developers to update versions of dependencies outside the default maui version is a feature not a bug.

**Comment by KieranDevvs** [comment]
> > This is because you're referencing it also in the android and ios  app.
> > If you reference it like this, it will work (only for Windows!)
> > ```
> > <ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.19041.0'">
> >    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230331000" />
> > </ItemGroup>
> > ```
> 
> I think this appears to be a bug because you have to do this, but this is as designed. When a new "major" version of wasdk comes out we will take some time before updating.
> 
> Versions 1.2 and 1.3 are considered "major"-ish versions by the wasdk team. So we don't want to go updating everyone as soon as there is a new version.
> 
> .NET 7 is based on 1.2 and .NET 8 is based on 1.3. Depending on feedback - such as this issue - we will investigate and consider updating the version that comes with the Maui install.
> 
> Regardless of everything and anything, having to specify the updated version of wasdk is a feature. Allowing developers to update versions of dependencies outside the default maui version is a feature not a bug.

I understand what you're saying but then the WindowsSDK patch needs to be applied to 1.2 in a minor update so that a minor revision for MAUI can be released?

Otherwise you're saying you cant fix a show stopper bug out of the box, until a major release of MAUI?

**Comment by creacity** [comment]
I am currently developing 1 MAUI Blazor Library project and 5 MAUI Blazor projects based on it.
I have no problem compiling for Android or debugging in Windows/Android environment.
However, when I want to compile as WindowsPackage, I run into the problem mentioned here; [microsoft-ui-xaml/issues/7846](https://github.com/microsoft/microsoft-ui-xaml/issues/7846)

So temporarily, I have to copy all the functions in the Library Project to all 5 projects. 
Unless I misunderstood from your answer, until .NET 8, the MAUI Library Project type is garbage.

**Comment by HelenMamalaki** [comment]
@mattleibow  In addition to this, here is another major issue fixed in 1.3 https://github.com/microsoft/microsoft-ui-xaml/issues/6291
Not having the onscreen keyboard appear on tapping into entry fields is major blocker for going to production. Last I heard .NET 8 is unstable for us to upgrade to the preview version. I will try the code block you posted.

**Comment by AnithaRaniT** [comment]
I'm also facing the issue, which is blocking me the production release. someone help me to overcome this issue. 

I'm unable to update the Microsoft.WindowsAppSDK version from 1.2.221109.1 to any higher version due to my project is blocking this from the update. PFA FYI

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/45478510/6d4ad289-cc06-4b88-94ec-b874e2d02a77)


I found one suggestion is that to add in the .csproj file is that,

`<ItemGroup Condition="'$(TargetFramework)'=='net7.0-windows10.0.17763'">
    <PackageReference Include="Microsoft.WindowsAppSDK" Version="1.3.230331000" />
  </ItemGroup>`


after adding that also not able to publish the app package for the windows platform. 


Here is my target platforms:
`	<TargetFrameworks>net7.0-ios;net7.0-android33.0</TargetFrameworks>
		<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net7.0-windows10.0.20348.0</TargetFrameworks>`


@IsmailHassani  Please provide some solution for this as this is high priority which is blocking from the production release.

Here is my VS version details. 

![image](https://github.com/microsoft/microsoft-ui-xaml/assets/45478510/d8edc03c-1c48-4580-8ecc-1ae3fd3bc1fc)


Please let me know if any details required

**Comment by IsmailHassani** [comment]
@AnithaRaniT Hi there, what is the error you're getting?
Because this adjustment should be done directly in your cs-proj  file.
Not from the nuget update tool.

**Comment by solomonfried** [comment]
I left this post at [https://github.com/microsoft/microsoft-ui-xaml/issues/7846#issuecomment-1747200566](https://github.com/microsoft/microsoft-ui-xaml/issues/7846#issuecomment-1747200566)

same issue. Cannot Publish a windows app due to the error 

"Improper project configuration: WindowsPackageType is set to None, but PublishAppxPackage is set to true."

Advice given was to upgrade to Windows SDK 1.3, but this is not allowed from the Nuget manager.

This issue of not being able to Publish a windows Maui App containing a Maui class library has been open for a year.
I would think this would be considered Critical!!!

**Comment by AnithaRaniT** [comment]
@solomonfried  You can publish your with help of CLI which is documented here: https://learn.microsoft.com/en-us/dotnet/maui/windows/deployment/publish-cli

Before that you need to change the TargetFrameworks for the windows as 
<TargetFrameworks Condition="$([MSBuild]::IsOSPlatform('windows'))">$(TargetFrameworks);net7.0-windows10.0.19041.0</TargetFrameworks>

and then follow the guide.

**Comment by strich** [comment]
This continues to be a serious issue on the default .NET 7 template. I'm lucky I stumbled upon this issue otherwise I'd have just abandoned MAUI.
